### PR TITLE
Create post mortem post for 03.08.23

### PR DESCRIPTION
**Pull request by hellcp-work**

### CODE REVIEW

**Review by danidoni — APPROVED** [review]

File: _posts/deployments/2023-08-03-post-mortem.md [hunk 1]
```diff
@@ -1,37 +1,47 @@
 ## Downtime on the afternoon of 3rd of August
 
-On 3. August, a few hours after a large migration performed within the maintenance window earlier that day, we got notified about database errors coming from one of the tables.
+On 3. August, a few hours after a large migration performed within the maintenance window earlier that day, we got notified about database errors coming from some of the tables.
 
 **Date**: 03.08.2023 
 
-**Impact**: Broken `project_log_entries` table
+**Impact**: Broken indexes on `project_log_entries`, `binary_releases`, `bs_request_actions` and `tokens` tables
```
Comment (danidoni) [inline _posts/deployments/2023-08-03-post-mortem.md:7]:
```suggestion
**Impact**: Broken indexes on `project_log_entries`, `binary_releases`, `bs_request_actions` and `tokens` tables.
```

File: _posts/deployments/2023-08-03-post-mortem.md [hunk 2]
```diff
@@ -1,37 +1,47 @@
...
 
-**Root Causes**: Database cluster VM run out of available space
+**Root Causes**: Database cluster VM run out of available space. Generation of table indexes did not work
```
Comment (danidoni) [inline _posts/deployments/2023-08-03-post-mortem.md:9]:
```suggestion
**Root Causes**: Database cluster VM run out of available space. Generation of table indexes did not work.
```

Comment (danidoni) [reply _posts/deployments/2023-08-03-post-mortem.md:9]: Also, generation as in creation?

File: _posts/deployments/2023-08-03-post-mortem.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+On 3. August, a few hours after a large migration performed within the maintenance window earlier that day, we got notified about database errors coming from some of the tables.
+
+**Date**: 03.08.2023 
+
+**Impact**: Broken indexes on `project_log_entries`, `binary_releases`, `bs_request_actions` and `tokens` tables
+
+**Root Causes**: Database cluster VM run out of available space. Generation of table indexes did not work
+
+**Trigger**: Morning deployment and migration from utf8mb3 to utf8mb4
```
Comment (danidoni) [inline _posts/deployments/2023-08-03-post-mortem.md:11]:
```suggestion
**Trigger**: Morning deployment and migration from utf8mb3 to utf8mb4.
```

File: _posts/deployments/2023-08-03-post-mortem.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+**Resolution**: The tables were dumped and restored from scratch
```
Comment (danidoni) [inline _posts/deployments/2023-08-03-post-mortem.md:13]:
```suggestion
**Resolution**: The tables were dumped and restored from scratch.
```

File: _posts/deployments/2023-08-03-post-mortem.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+**Detection**: We were informed by BuildOPS who got notified via their monitoring
```
Comment (danidoni) [inline _posts/deployments/2023-08-03-post-mortem.md:15]:
```suggestion
**Detection**: We were informed by BuildOPS who got notified via their monitoring.
```

File: _posts/deployments/2023-08-03-post-mortem.md [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+**What went wrong?**
+
+* The current way we deploy with migration does not log the progress of migrations or inform us about things happening in real time
+* We don't know about errors coming from the database if there are any unless somebody tells us
+* We did not communicate this migration with the BuildOPS in advance to warn them of any potential fallout
+
+**Where we got lucky?**
+
+Only four tables ended up being affected
```
Comment (danidoni) [inline _posts/deployments/2023-08-03-post-mortem.md:30]:
```suggestion
The rest of the migration except for those tables.

**What went wrong?**

* The current way we deploy with migration does not log the progress of migrations or inform us about things happening in real time.
* We don't know about errors coming from the database if there are any unless somebody tells us.
* We did not communicate this migration with the BuildOPS in advance to warn them of any potential fallout.

**Where we got lucky?**

Only four tables ended up being affected.
```

**Review by hennevogel — CHANGES_REQUESTED** [review]
you are missing [front matter](https://jekyllrb.com/docs/front-matter/)

File: _posts/deployments/2023-08-03-post-mortem.md [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+category: deployments
+---
+
+## Downtime on the afternoon of 3rd of August
+
+On 3. August, a few hours after a large migration performed within the maintenance window earlier that day, we got notified about database errors coming from some of the tables.
+
+**Date**: 03.08.2023 
+
+**Impact**: Broken indexes on `project_log_entries`, `binary_releases`, `bs_request_actions` and `tokens` tables.
```
Comment (hennevogel) [inline _posts/deployments/2023-08-03-post-mortem.md:13]: Impact for customers not for the involved software.

File: _posts/deployments/2023-08-03-post-mortem.md [hunk 8]
```diff
@@ -0,0 +1,54 @@
+---
+layout: post
+title: "Post-mortem: Database Cluster Crashes"
+category: deployments
+---
+
+## Downtime on the afternoon of 3rd of August
+
+On 3. August, a few hours after a large migration performed within the maintenance window earlier that day, we got notified about database errors coming from some of the tables.
```
Comment (hennevogel) [inline _posts/deployments/2023-08-03-post-mortem.md:9]: We usually tell people how it affected them so they can identify what we are talking about. Remember we write this for everyone else, not for us.

File: _posts/deployments/2023-08-03-post-mortem.md [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+**Trigger**: Morning deployment and migration from utf8mb3 to utf8mb4.
+
+**Resolution**: The tables were dumped and restored from scratch.
+
+**Detection**: We were informed by BuildOPS who got notified via their monitoring.
+
+## Lessons Learned
+**What went well?**
+
+The rest of the migration except for those tables.
```
Comment (hennevogel) [inline _posts/deployments/2023-08-03-post-mortem.md:26]: Not sure what you want to say? That the change we made was good?

File: _posts/deployments/2023-08-03-post-mortem.md [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+
+**Impact**: Broken indexes on `project_log_entries`, `binary_releases`, `bs_request_actions` and `tokens` tables.
+
+**Root Causes**: Database cluster VM run out of available space. Generation of table indexes did not work.
+
+**Trigger**: Morning deployment and migration from utf8mb3 to utf8mb4.
+
+**Resolution**: The tables were dumped and restored from scratch.
+
+**Detection**: We were informed by BuildOPS who got notified via their monitoring.
```
Comment (hennevogel) [inline _posts/deployments/2023-08-03-post-mortem.md:21]:
```suggestion
**Detection**: Our database admins got notified via their monitoring.
```

File: _posts/deployments/2023-08-03-post-mortem.md [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+
+## Downtime on the afternoon of 3rd of August
+
+On 3. August, a few hours after a large migration performed within the maintenance window earlier that day, we got notified about database errors coming from some of the tables.
+
+**Date**: 03.08.2023 
+
+**Impact**: Broken indexes on `project_log_entries`, `binary_releases`, `bs_request_actions` and `tokens` tables.
+
+**Root Causes**: Database cluster VM run out of available space. Generation of table indexes did not work.
```
Comment (hennevogel) [inline _posts/deployments/2023-08-03-post-mortem.md:15]:
```suggestion
**Root Causes**: Our database cluster ran out of available space during a large schema/data migration ([#14597 - Migrate the remaining database tables and columns to utf8mb4](https://github.com/openSUSE/open-build-service/pull/14597))
```

File: _posts/deployments/2023-08-03-post-mortem.md [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+**Detection**: We were informed by BuildOPS who got notified via their monitoring.
+
+## Lessons Learned
+**What went well?**
+
+The rest of the migration except for those tables.
+
+**What went wrong?**
+
+* The current way we deploy with migration does not log the progress of migrations or inform us about things happening in real time.
```
Comment (hennevogel) [inline _posts/deployments/2023-08-03-post-mortem.md:30]: Where is the action item to change this?

File: _posts/deployments/2023-08-03-post-mortem.md [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+* We don't know about errors coming from the database if there are any unless somebody tells us.
```
Comment (hennevogel) [inline _posts/deployments/2023-08-03-post-mortem.md:31]: Not sure what you want to say with this. Can you explain?

File: _posts/deployments/2023-08-03-post-mortem.md [hunk 14]
```diff
@@ -0,0 +1,54 @@
...
+* We did not communicate this migration with the BuildOPS in advance to warn them of any potential fallout.
```
Comment (hennevogel) [inline _posts/deployments/2023-08-03-post-mortem.md:32]:
```suggestion
* We did not communicate this migration with our database admins in advance to make them aware of potential fallout.
```

File: _posts/deployments/2023-08-03-post-mortem.md [hunk 15]
```diff
@@ -0,0 +1,54 @@
...
+
+**Where we got lucky?**
+
+* Only four tables ended up being affected.
+* There were people around who helped us in getting the database back to usable state.
```
Comment (hennevogel) [inline _posts/deployments/2023-08-03-post-mortem.md:37]:
```suggestion
* Our database admins where around to help us in getting the database back to usable state.
```

File: _posts/deployments/2023-08-03-post-mortem.md [hunk 16]
```diff
@@ -0,0 +1,54 @@
...
+
+## Timeline (CEST)
+- *09:03* Started the deployment with the migration
+- *09:26* Ended the deployment
+- *09:31* First recorded error in the index of `project_log_entries` table
+- *13:37* Database cluster crashes
```
Comment (hennevogel) [inline _posts/deployments/2023-08-03-post-mortem.md:43]: Hm, how did we survive this gap between 9 and 1?

**Review by hennevogel — CHANGES_REQUESTED** [review]

File: _posts/deployments/2023-08-03-post-mortem.md (hunk 9 above)
Comment (hennevogel) [reply _posts/deployments/2023-08-03-post-mortem.md:26]: Yeah but which table was affected or not had little to do with the migration right? It's a bits misleading, as if the migration would be the cause...

File: _posts/deployments/2023-08-03-post-mortem.md (hunk 13 above)
Comment (hennevogel) [reply _posts/deployments/2023-08-03-post-mortem.md:31]: Well, no wonder right? We don't maintain the database cluster :)

File: _posts/deployments/2023-08-03-post-mortem.md (hunk 9 above)
Comment (hennevogel) [reply _posts/deployments/2023-08-03-post-mortem.md:26]: I wouldn't mention it at all. It has nothing really to do with the incident...

File: _posts/deployments/2023-08-03-post-mortem.md (hunk 12 above)
Comment (hennevogel) [reply _posts/deployments/2023-08-03-post-mortem.md:30]: SRE I would say https://trello.com/b/Qx7osByV

File: _posts/deployments/2023-08-03-post-mortem.md (hunk 13 above)
Comment (hennevogel) [reply _posts/deployments/2023-08-03-post-mortem.md:31]: For sure not but you are capable of rewording :)

File: _posts/deployments/2023-08-03-post-mortem.md [hunk 17]
```diff
@@ -0,0 +1,53 @@
...
+**Detection**: Our database admins got notified via their monitoring.
+
+## Lessons Learned
+**What went well?**
+
+* We learned about the database crash soon after it happened.
+
+**What went wrong?**
+
+* The current way we deploy with migration does not log the progress of migrations or inform us about things happening in real time.
```
Comment (hennevogel) [inline _posts/deployments/2023-08-03-post-mortem.md:30]:
```suggestion
* The current way we deploy with migration does not log the progress of migrations or inform us about things happening in real time ([improvement card](https://trello.com/c/iQKJUUaS)).
```

**Review by hennevogel — CHANGES_REQUESTED** [review]

**Review by hennevogel — APPROVED** [review]

**Pull request merged**